doc/introduction/measurements.rst

```diff
@@ -300,16 +301,28 @@ The decorator syntax applies equally well:
     def qnode(x, y):
         (...)
 
+    @qml.defer_measurements
+    @qml.qnode(dev)
+    def qnode(x, y):
+        (...)
+
+.. note::
+
+    The deferred measurements principle requires an additional wire, or qubit, for each mid-circuit
+    measurement, limiting the number of measurements that can be used both on classical simulators
+    and quantum hardware. The one-shot transform below does not have this limitation, but has
+    computational cost that scale with the used number of shots.
```

Comment:
We should definitely mention this requirement, in my opinion. Also because we reference it in the one-shot transform section below.